caddyhttp/proxy/reverseproxy.go

```diff
@@ -200,7 +200,7 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, outreq *http.Request,
 		respUpdateFn(res)
 	}
 	if res.StatusCode == http.StatusSwitchingProtocols && strings.ToLower(res.Header.Get("Upgrade")) == "websocket" {
-		res.Body.Close()
+		defer res.Body.Close()
```

Comment:
This Close() is also the first line of the `else` condition below; let's remove both of these and just have a single one above the if statements; probably on line 198, right after the error check. And yes, keep the defer.

(In fact, what I'm basically asking for here is to revert to what it was before the PR I linked to above.)